internal/provider/cluster_resource.go

```diff
@@ -123,7 +122,7 @@ func (r *clusterResource) Schema(
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.UseStateForUnknown(),
 				},
-				Description: "Major version of CockroachDB running on the cluster.",
+				MarkdownDescription: "Major version of CockroachDB running on the cluster. This value can be used to orchestrate version upgrades. Supported for ADVANCED and STANDARD clusters (when `serverless.upgrade_type` set to 'MANUAL').",
```

Comment:
@kathancox thanks in advanced for looking over this. I'm not sure it's enough or whether we have other dedicated documentation on updating the cockroach version with terraform.  I could link out to an external doc if we have one.